### PR TITLE
use regex in docker version check to specifically target semantic ver…

### PR DESCRIPTION
**Pull request by adamsachs**
Closes n/a

Hopefully should patch up e2e failures on `main`, while also keeping the desired functionality in the version check...

Drafted this in a pinch, very open to suggestions!

### Code Changes

* handle some uglier docker version outputs by targeting a semver number directly with a regex

### Steps to Confirm

* works locally with at "clean" docker version (very basic regression test)
* fixes e2e test in CI runner which seemed to have an _uglier_ docker version

### Pre-Merge Checklist

* [x] All CI Pipelines Succeeded
* Documentation:
  * [ ] documentation complete, [PR opened in fidesdocs](https://github.com/ethyca/fidesdocs/pulls)
  * [ ] documentation [issue created in fidesdocs](https://github.com/ethyca/fidesdocs/issues/new/choose)
* [x] Issue Requirements are Met
* [ ] Relevant Follow-Up Issues Created
* [ ] Update `CHANGELOG.md`
* [ ] For API changes, the [Postman collection](https://github.com/ethyca/fides/blob/main/docs/fides/docs/development/postman/Fides.postman_collection.json) has been updated

### CODE REVIEW

File: src/fides/core/deploy.py [hunk 1]
```diff
@@ -89,9 +90,17 @@ def check_docker_version(
             raise DockerCheckException(
                 "Could not determine Docker version from 'docker' commands. Please ensure that Docker is installed and running and try again."
             )
-        parsed_version = (
+        stripped_version = (
             raw_version.stdout.decode("utf-8").rstrip("\n").replace("'", "")
         )
+        parsed_version = re.match(r"((\d+\.)+\d+)", stripped_version)
```
Comment (adamsachs) [inline src/fides/core/deploy.py:96]: used `match` instead of `search` because i think we can assume the version number is the beginning of the string, especially after stripping it above???

Comment (NevilleS) [reply src/fides/core/deploy.py:96]: Eh... if there's just a single preceding space ` ` you'll be left with nothing!

The `rstrip("\n")` seems to just strip trailing newlines, not touch whitespace at all.

TL;DR: I would chain one more `.strip()` to the parsing

**Review by NevilleS — COMMENTED** [review]
Feels a bit brittle still...

File: src/fides/core/deploy.py (hunk 1 above)
Comment (adamsachs) [reply src/fides/core/deploy.py:96]: ooh, right didn't pay close enough attention to what we were stripping. i'll do as you suggest, that feels reasonable.

**Review by NevilleS — APPROVED** [review]
LGTM!

**Comment by ThomasLaPiana** [comment]
@adamsachs for my own curiosity here, what was the output that broke this? I was thinking the formatting I used would've been quite clear/robust, but apparently not 🤔

**Comment by adamsachs** [comment]
@ThomasLaPiana what @NevilleS spotted this in the [nox output from that CI job](https://github.com/ethyca/fides/actions/runs/4885622167/jobs/8719901123#step:5:9):

```
nox > Running session fides_env(test)
nox > Creating virtual environment (virtualenv) using python3 in .nox/fides_env-test
Docker version 20.10.24+azure-1, build 297e1284d3bd092e9bc96076c3ddc4bb33f8c7ab
```
so we assumed something along the likes of `20.10.24+azure-1` was still getting returned to your docker version command in the CI runtime. i suppose we could have (or still could) confirm more precisely by adding in some additional logging 🤔

**Comment by ThomasLaPiana** [comment]
Oh that makes sense 🤔 I feel like this could be cleaned up further still but honestly the value isn't there right now, we can revisit later 🙂

**Comment by ThomasLaPiana** [comment]
Thanks for the rescue there!
